platform/linuxbsd/freedesktop_portal_desktop.cpp

```diff
@@ -617,6 +749,263 @@ void FreeDesktopPortalDesktop::_system_theme_changed_callback() {
 	}
 }
 
+void FreeDesktopPortalDesktop::_status_notifier_item_unregister(DBusConnection *connection, void *user_data) {
+}
+
+DBusHandlerResult FreeDesktopPortalDesktop::_status_notifier_item_handle_message(DBusConnection *connection, DBusMessage *message, void *user_data) {
+	FreeDesktopPortalDesktop *portal = (FreeDesktopPortalDesktop *)user_data;
+	ERR_FAIL_NULL_V(portal, DBUS_HANDLER_RESULT_HANDLED);
+
+	String interface = dbus_message_get_interface(message);
+	String member = dbus_message_get_member(message);
+
+	const char *destination_ptr = dbus_message_get_destination(message);
+	ERR_FAIL_NULL_V(destination_ptr, DBUS_HANDLER_RESULT_HANDLED);
+
+	String destination;
+	destination.parse_utf8(destination_ptr);
+
+	ERR_FAIL_COND_V(!portal->indicator_id_map.has(destination), DBUS_HANDLER_RESULT_HANDLED);
+	DisplayServer::IndicatorID id = portal->indicator_id_map[destination];
+
+	ERR_FAIL_COND_V(!portal->indicators.has(id), DBUS_HANDLER_RESULT_HANDLED);
+	StatusNotifierItem &item = portal->indicators[id];
+
+	if (dbus_message_is_method_call(message, BUS_INTERFACE_STATUS_NOTIFIER_ITEM, "Activate")) {
```

Comment:
Probably should also handle `SecondaryActivate` for other mouse button clicks (not sure if there is a way to determine exact mouse button, but it's probably OK to call cb with MIDDLE button for it).

Reply:
I think activation callback on macOS/Windows have position and mouse button arguments, this events should have position as well and we can set mouse button to RIGHT for primary and MIDDLE (or LEFT) for secondary (mouse button is not precise on Windows as well, but primary activation pretend to be right click).